shopify_app.gemspec

```diff
@@ -8,9 +8,7 @@ Gem::Specification.new do |s|
   s.author      = "Shopify"
   s.summary     = %q{This gem is used to get quickly started with the Shopify API}
 
-  s.add_dependency('rails', '>= 4.2.6', '< 5.0')
```

Comment:
@swalkinshaw, I don't see this particular post very relevant in our case, as it looks like Yehuda writes in the context of duplication in single gemspec and single Gemfile, which is obviously is not our case.

Ironically - duplication is the very thing I intended to avoid by explicitly putting differentiating versions of dependency in different Gemspecs.

Another idea - not putting dependency version restrictions into gemspec at all.

And yet another idea-- not using same Gemfiles files for different purposes looks appealing:
```
Gemfile
Gemfile.ci-rails-master
Gemfile.ci-rails-5.0-stable
Gemfile.ci-rails-4.2-stable
```
Another crazy idea -- having both separate Gemfiles and gemspec files for CI matrix:
Sure -- overriding dependencies in Gemspec may be a valid approach, I just avoided it
Anyway, core of the issue I believe is having multiple Gemfiles, for CI matrix, which is a reflection of maintenance/support and versioning policy (of which I have no knowledge at all)

As I mentioned before, my hypothesis about source of shopify_api issue is a use of wrong Gemfile during gem publishing process